app/Services/GitHub.php

```diff
@@ -41,6 +47,55 @@ public function isCompanySponsor(string $username): bool
         )->values()->isNotEmpty();
     }
 
+    /**
+     * Validates the user received from Github and links our user to it
+     *
+     * @return array<string, mixed> list of validation errors
+     */
+    public function linkGitHubUser(?string $githubUsername, User $user, Request $request): array
+    {
+        try {
+            $validated = $this->validateGitHubUsername($githubUsername);
+        } catch (ValidationException $e) {
+            return $e->errors();
+        }
+
+        $user->update($validated);
+
+        SyncVerifiedUser::dispatchSync($user);
+
+        $user = type($user->fresh())->as(User::class);
+
+        $user->is_verified
+            ? $request->session()->flash('flash-message', 'Your GitHub account has been connected and you are now verified.')
+            : $request->session()->flash('flash-message', 'Your GitHub account has been connected.');
+
+        if (! $user->is_uploaded_avatar) {
+            UpdateUserAvatar::dispatch(
+                user: $user,
+                service: 'github',
+            );
+        }
+
+        return [];
+    }
+
+    /**
+     * @return array<string, mixed>
+     *
+     * @throws ValidationException
+     */
+    private function validateGitHubUsername(?string $githubUsername): array
+    {
+        return Validator::validate([
```

Comment:
I extracted this function because PhpStan seemed to be unable to guess that this piece of code could throw a ValidationException